### ISSUE TITLE
Lots of timeouts and "No route to host" errors

### ISSUE THREAD

**Issue by johnhamelink** [nginx-proxy/nginx-proxy#191]
Hi there,

I'm using nginx-proxy to proxy a connection between the outside world and a rails app using puma. However the connection is very intermittent and I get a lot of the following messages:

```
2015/07/02 15:41:31 [error] 48#48: *4484 connect() failed (113: No route to host) while connecting to upstream, client: 82.23.149.132, server: api.example.com, request: "GET /users/devices/mine.json HTTP/1.1", upstream: "http://172.17.1.13:3000/users/devices/mine.json", host: "blah.example.com"
```

My docker-compose.yml looks like this (private images names have been redacted):

``` yaml
web:
  image: example/rails-app:latest
  ports:
    - "3000:3000"
  links:
    - redis
  volumes:
    - "./log:/var/www/log"
    - "./assets:/var/www/public/assets"
  env_file: .env
  restart: always

nginx:
  image: jwilder/nginx-proxy:latest
  ports:
    - "80:80"
  volumes:
    - /var/run:/tmp
  restart: always

sidekiq:
  image: example/rails-app:latest
  links:
    - redis
  volumes:
    - "./sidekiq_log:/var/www/log"
  env_file: .env
  restart: always
  command: bundle exec sidekiq

redis:
  image: redis
  ports:
    - "6379:6379"
  volumes:
    - "./redis_data:/data"

rpush:
  image: example/rails-app:latest
  volumes:
    - "./rpush_log:/var/www/log"
  env_file: .env
  restart: always
  command: bundle exec rpush start -e production -f

logs:
  image: progrium/logspout
  restart: always
  hostname: example-app
  volumes:
    - /var/run/docker.sock:/tmp/docker.sock
  command: syslog://logs3.papertrailapp.com:19957
  restart: always
```

**Comment by matheuscmpm** [comment]
No responses at all about this issue? I am having quite few returns just like this one as well. Any ideia of what is causing? I started the debug process today - hope can find what is cracking.

**Comment by matheuscmpm** [comment]
Saw here that this could be a problem related to our docker version: https://github.com/jwilder/nginx-proxy/issues/1348

Just updated mine to version 19 - still trying to reproduce after this upgrade. I was using v18 before.

**Comment by DigitalJosee** [comment]
Actually updating docker didn't fixed my problem, the real problem was with the number of dockers, any IP equal or higher than 172.18.0.14 are not working. Since I had exactly 14 containers, I had to disable one and now all of them are working fine.

**Comment by matheuscmpm** [comment]
> Actually updating docker didn't fixed my problem, the real problem was with the number of dockers, any IP equal or higher than 172.18.0.14 are not working. Since I had exactly 14 containers, I had to disable one and now all of them are working fine.

Maybe you were getting an out of memory error inside your containers. My server is running with 143+ containers, 353 volumes.. and this error was raised not that often. After updating my docker, I got a false positive - I thought that could had fixed my problem - but it didn't. 

I also created a swap partition - trying to prevent this error in case of out of memory - but still - the problem still exists. 

Still trying to figure this out...
